### PR TITLE
Add Microsoft Graph auth provider

### PR DESCRIPTION
**Pull request by synth3tk**
My PR is a:
🚀 New feature

Main update on the:
Plugin

This will add the Microsoft Graph as a provider option, which covers personal accounts as well as Office 365 accounts. This resolves the request in #1385.

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 1]
```diff
@@ -172,6 +172,40 @@ const getProfile = async (provider, query, callback) => {
...
+              '[version]/{endpoint}': {
+                '__path': {
+                  'alias': '__default',
+                  'version': 'v1.0'
+                }
+              }
+            }
+          }
+        }
+      });
```
Comment (synth3tk) [inline packages/strapi-plugin-users-permissions/services/Providers.js]: @derrickmehaffy Here's the part I had to add for Microsoft. You'll most likely need to do the same for Discord.

Comment (derrickmehaffy) [reply packages/strapi-plugin-users-permissions/services/Providers.js]: Thanks @synth3tk I'll jump on this :+1:

**Comment by lauriejim** [comment]
Hello @synth3tk, thank you for this PR ! Have to make some test. If anyone has already test it. Feel free  to send your feedback.

**Comment by lauriejim** [comment]
As comment on Twitch provider PR. Can you add documentation please.

**Pull request merged**